recipes/zbar/all/conanfile.py

```diff
@@ -87,69 +85,60 @@ def build_requirements(self):
-                "--enable-video={}".format(yes_no(self.options.with_video)),
-                "--with-imagemagick={}".format(yes_no(self.options.with_imagemagick)),
-                "--with-gtk={}".format(yes_no(self.options.with_gtk)),
-                "--with-qt={}".format(yes_no(self.options.with_qt)),
-                "--with-python={}".format(yes_no(self.options.with_python_bindings)),
-                "--with-x={}".format(yes_no(self.options.with_x)),
-                "--with-xshm={}".format(yes_no(self.options.with_xshm)),
-                "--with-xv={}".format(yes_no(self.options.with_xv)),
-                "--with-jpeg={}".format(yes_no(self.options.with_jpeg)),
-                "--enable-pthread={}".format(yes_no(self.options.enable_pthread)),
-            ]
-            if self.settings.os == "Macos" and self.settings.arch == "armv8":
-               # ./libtool: eval: line 961: syntax error near unexpected token `|'
-                args.append("NM=nm")
-            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        yes_no = lambda v: "yes" if v else "no"
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.extend([
+            f"--enable-video={yes_no(self.options.with_video)}",
+            f"--with-imagemagick={yes_no(self.options.with_imagemagick)}",
+            f"--with-gtk={yes_no(self.options.with_gtk)}",
+            f"--with-qt={yes_no(self.options.with_qt)}",
+            f"--with-python={yes_no(self.options.with_python_bindings)}",
+            f"--with-x={yes_no(self.options.with_x)}",
+            f"--with-xshm={yes_no(self.options.with_xshm)}",
+            f"--with-xv={yes_no(self.options.with_xv)}",
+            f"--with-jpeg={yes_no(self.options.with_jpeg)}",
+            f"--enable-pthread={yes_no(self.options.enable_pthread)}",
+        ])
+        tc.generate()
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # ./libtool: eval: line 961: syntax error near unexpected token `|'
+            env = Environment()
+            env.define("NM", "nm")
+            env.vars(self).save_script("conanbuild_macos_nm")
+
 
     def build(self):
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
-                    os.path.join(self._source_subfolder, "config", "config.sub"))
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
-                    os.path.join(self._source_subfolder, "config", "config.guess"))
+        copy(self, "config.sub", src=self.source_folder, dst=os.path.join(self.source_folder, "config"))
+        copy(self, "config.guess", src=self.source_folder, dst=os.path.join(self.source_folder, "config"))
 
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
```

Comment:
```suggestion
        autotools = Autotools(self)
        if Version(self.version) > "0.10":
            autotools.autoreconf()
```
The version 0.23 is available on Github, but there is no configure file available.